helix-vcs/src/git.rs

```diff
@@ -61,7 +65,30 @@ impl DiffProvider for Git {
         let file_oid = find_file_in_commit(&repo, &head, file)?;
 
         let file_object = repo.find_object(file_oid).ok()?;
-        Some(file_object.detach().data)
+        let mut data = file_object.detach().data;
+        // convert LF to CRLF if configured to avoid showing every line as changed
+        if repo
+            .config_snapshot()
+            .boolean("core.autocrlf")
+            .unwrap_or(false)
+        {
+            let mut normalized_file = Vec::with_capacity(data.len());
+            let mut at_cr = false;
+            for &byte in &data {
+                if byte == b'\n' {
+                    // if this is a LF instead of a CRLF (last byte was not a CR)
+                    // insert a new CR to generate a CRLF
+                    if !take(&mut at_cr) {
```

Comment:
This is the first time I have seen this being used to turn off a boolean after checking its value, and it's neat despite maybe being a little implicit.

Reply:
I agree: I think it would be clearer to set `at_cr` separately from reading it. The `!` plus the `take` makes this pretty dense.  Plus I think it's non-obvious that `false` is the `Default` of `bool` which can make this look surprising

Reply:
I removed the use of `take`